src/util/metadata/side_metadata/global.rs

```diff
@@ -154,160 +155,52 @@ impl SideMetadataSpec {
-            // Clippy wants to move this if block up as a else-if block. But I think this is logically more clear. So disable the clippy warning.
-            #[allow(clippy::collapsible_else_if)]
-            if forwards {
-                // update bits in the first byte
-                if Self::iterate_meta_bits(
-                    meta_start_addr,
-                    meta_start_bit,
-                    meta_start_addr + 1usize,
-                    0,
-                    forwards,
-                    visit_bytes,
-                    visit_bits,
-                ) {
-                    return true;
-                }
-                // update bytes in the middle
-                if Self::iterate_meta_bits(
-                    meta_start_addr + 1usize,
-                    0,
-                    meta_end_addr,
-                    0,
-                    forwards,
-                    visit_bytes,
-                    visit_bits,
-                ) {
-                    return true;
-                }
-                // update bits in the last byte
-                if Self::iterate_meta_bits(
-                    meta_end_addr,
-                    0,
-                    meta_end_addr,
-                    meta_end_bit,
-                    forwards,
-                    visit_bytes,
-                    visit_bits,
-                ) {
-                    return true;
-                }
-                false
-            } else {
-                // update bits in the last byte
-                if Self::iterate_meta_bits(
-                    meta_end_addr,
-                    0,
-                    meta_end_addr,
-                    meta_end_bit,
-                    forwards,
-                    visit_bytes,
-                    visit_bits,
-                ) {
-                    return true;
-                }
-                // update bytes in the middle
-                if Self::iterate_meta_bits(
-                    meta_start_addr + 1usize,
-                    0,
-                    meta_end_addr,
-                    0,
-                    forwards,
-                    visit_bytes,
-                    visit_bits,
-                ) {
-                    return true;
-                }
-                // update bits in the first byte
-                if Self::iterate_meta_bits(
-                    meta_start_addr,
-                    meta_start_bit,
-                    meta_start_addr + 1usize,
-                    0,
-                    forwards,
-                    visit_bytes,
-                    visit_bits,
-                ) {
-                    return true;
-                }
-                false
-            }
-        }
     }
 
-    /// This method is used for bulk zeroing side metadata for a data address range.
-    pub(super) fn zero_meta_bits(
+    /// Expose `zero_meta_bits` when running `cargo bench`.
+    #[cfg(feature = "bench")]
+    pub fn bench_zero_meta_bits(
```

Comment:
We may move these functions to a different module, like `util::test_util`, so we don't need to worry about duplicate function names. Also they won't appear as a disruption when we read the code.

Reply:
I think it's better to just add a top-level module `crate::testing` because they are not really "utilities for testing".  They are just the functions to be tested.

Reply:
I moved it to `util::test_private`.  It does not appear on the first page of the cargo docs.